### DISCUSSION TITLE
AudioTools.h compilation issues on Arduino Nano 33

### DISCUSSION THREAD

**Discussion by magicmomo123** [pschatzmann/arduino-audio-tools#788] [Q&A] [answered]
Hello Phil, 

Just had a quick problem using some of the structs defined in AudioTools.h, specifically the I2SStream and the URLStream. When I switch my board to ESP32 it compiles fine, but I am having an issue compiling with an arduino nano. 

Any help would be appreciated.

**Comment by pschatzmann** [comment, marked as answer]
On each environment you only get what is supported: The nano does not have any I2S support and no Wifi!
You could try to do some output with some low sample rates using PWM, but maybe you best use it with a VS1053...

**Reply by magicmomo123** [reply]
Sorry, should have been a bit more specific, I am using the Nano 33 IoT board, which should have both Wifi and I2S support. I will edit the title.

**Reply by pschatzmann** [reply]
I never managed to get I2S working on that board...
Did you ?

**Reply by magicmomo123** [reply]
Got it, no problem! I'll wait for my esp32 dev kit to come in. Only using an arduino nano at the moment because my last dev kit broke :).

**Reply by magicmomo123** [reply]
No not yet, but I will try to come up with something.

**Reply by pschatzmann** [reply]
I also was under the impression that this is supporting BLE only. 
Can you confirm that Wifi is also working there ?
